INSS.EIIR.DataSync.Infrastructure/Sink/XML/XMLSink.cs

```diff
@@ -1,27 +1,183 @@
+            }
+
+            _blobServiceClient = new BlobServiceClient(_blobConnectionString);
+            _containerClient = _blobServiceClient.GetBlobContainerClient(_blobContainerName);
+
+            //IExtractRepository required to get filename and update extract availability though comes with high overhead
+            _eiirRepository = extractRepository;
+        }
+
+        public async Task Start() {
+
+            var extractJob = _eiirRepository.GetExtractAvailable();
+
+            //Following section could move to beginning of getting datasource, its checking pre-conditions
+            //Which historically down before fetching data...not after
+            #region Move to ISCIS data source?
+            var today = DateOnly.FromDateTime(DateTime.Now);
+
+            var extractjobError = $"ExtractJob not found for today [{today}], snapshot has not run";
+            var snapshotError = $"Snapshot has not yet run today [{today}]";
+
+            if (extractJob == null)
+            {
+                throw new Exception(extractjobError);
+            }
+
+            if (extractJob.SnapshotCompleted?.ToLowerInvariant() == "n")
+            {
+                throw new Exception(snapshotError);
+            }
+
+            if (extractJob.ExtractCompleted?.ToLowerInvariant() == "y")
+            {
+                throw new Exception($"Subscriber xml / zip file creation has already ran successfully on [{today}]");
+            }
+            #endregion Move to ISCIS data source?
+
+            _extractId = extractJob.ExtractId;
+            _fileName = extractJob.ExtractFilename;
+            _xmlStream = new MemoryStream();
+
+            WriteIirHeaderToStream();
+
+            return; 
+        }
+
+        private void WriteIirHeaderToStream()
+        {
+            IirXMLWriterHelper.WriteIirHeaderToStream(ref _xmlStream);
+        }
+
+        private void WriteIirFoorterToStream()
+        {
+            IirXMLWriterHelper.WriteIirFooterToStream(ref _xmlStream);
+        }
 
         public async Task<SinkCompleteResponse> Complete()
         {
-            throw new NotImplementedException();
+            WriteIirFoorterToStream();
+
+            //Write out remainder of stream
+            await WriteStreamToBlob(true);
+
+            //Create Zip file
+            _eiirRepository.UpdateExtractAvailable();
+
+            _xmlStream.Close();
+            
+            return new SinkCompleteResponse() { IsError = false };
         }
 
         public async Task<DataSinkResponse> Sink(InsolventIndividualRegisterModel model)
         {
-            throw new NotImplementedException();
+            var resp = new DataSinkResponse() { IsError = false };
+
+            try 
+            {
+                WriteIirRecordToStream(model, ref _xmlStream);
+                _recordCount++;  
+
+            }
+            catch (Exception ex)
+            {
+                //Catch any known exception in here to put in respone
```

Comment:
Intention in these parts is to put in something when we find it? Fine with that - we should remove the try/catch entirely if we don't find anything in the fullness of time.